Test_Is_subtree.java

```diff
@@ -0,0 +1,75 @@
+package Week6;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class Test_Is_subtree {
+
+	@Test
+	void test_NullTrees() {
+		Is_Subtree s_tree=new Is_Subtree();
+		assertEquals(true,s_tree.isSubtree(null,null));
+	}
+	void test_s_null() {
+		Is_Subtree s_tree=new Is_Subtree();
+		TreeNode t_root=new TreeNode(2);
+		assertEquals(false,s_tree.isSubtree(null,t_root));		
+	}
+	void test_t_null() {
+		Is_Subtree s_tree=new Is_Subtree();
+		TreeNode s_root=new TreeNode(2);
+		assertEquals(false,s_tree.isSubtree(s_root,null));		
+	}
+	void test_ValidCase1() {
+		Is_Subtree s_tree=new Is_Subtree();
+		TreeNode s_root=new TreeNode(1);
+		s_root.left=new TreeNode(1);
+		TreeNode t_root=new TreeNode(1);
+		assertEquals(true,s_tree.isSubtree(s_root,t_root));
+	}
+	void test_ValidCase_False1() {

```

Comment:
Good tests overall. Just so you know though, test_ValidCase_False1 and test_ValidCase_False2 are actually testing the same thing. Maintaining tests can be a cumbersome and hard thing so at the end of the day you want the least number of valid tests possible, so that way when you make a change, only one test fails for exactly the change you made instead of multiple. This concept of tests doing the same thing is called "Equivalency Classes" and is an important thing to realize when writing tests; if two tests are essentially testing the same case (in this case, valid trees that are not subtrees), then we should only need one test that tests them